addons/website/static/src/snippets/s_searchbar/options.js

```diff
@@ -0,0 +1,38 @@
+odoo.define('website.s_searchbar_options', function (require) {
+'use strict';
+
+const options = require('web_editor.snippets.options');
+
+const searchBarOptions = options.Class.extend({
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    setSearchType: function (previewMode, widgetValue, params) {
+        const form = this.$target.parents('form');
+        form.attr('action', params.formAction);
+
+        if (!previewMode) {
+            this.trigger_up('snippet_edition_request', {exec: () => {
+                const widget = this._requestUserValueWidgets('order_opt')[0];
+                const orderBy = widget.getValue("selectDataAttribute");
+                const orderSelect = this.$el.find("we-select[data-name='order_opt']");
+                const order = orderSelect.find("we-button[data-select-data-attribute='" + orderBy + "']")[0];
+                if (order.classList.contains("d-none")) {
```

Comment:
What is the purpose of this part in the code (in which case you get hidden selected order options) ?

Reply:
For example, you first set `Search within` to `Blog`, this makes date-based `Order By` available.
Imagine you picked `Order By`: `date (new to old)`, if you then change `Search within` to `Products` this `Order By` does not exist anymore.
In the case the previously selected `Order by` is not available for the current `Search within`, this part of the code makes sure the first available `Order by` is selected.